### PR TITLE
[17.0][MIG] account_reconcile_oca: Migration to 17.0

### PR DESCRIPTION
**Pull request by etobella**
@norlinhenrik @JordiBForgeFlow 

TODO:

- [x] Migrate bank reconciliation widget
- [x] Migrate reconciliaion widget of account move lines
- [x] Move Code removed from Odoo EE

### CODE REVIEW

**Comment by norlinhenrik** [comment]
@etobella Thank you for this new PR :) The bank reconciliation works well.

I don't have the skills to migrate the reconciliaion widget of account move lines.

**Comment by etobella** [comment]
I will do it eventually, don't worry :wink:

**Comment by etobella** [comment]
@norlinhenrik it was a small fix. it is fixed now

**Review by norlinhenrik — APPROVED** [review]
Thank you @etobella 👍

File: account_reconcile_oca/static/src/js/widgets/reconcile_chatter_field.esm.js
Comment (tien-ld) [inline account_reconcile_oca/static/src/js/widgets/reconcile_chatter_field.esm.js]: please remove this

File: account_reconcile_oca/models/account_bank_statement_line.py
Comment (tien-ld) [inline account_reconcile_oca/models/account_bank_statement_line.py:673]: please add self.ensure_one()

File: account_reconcile_oca/views/account_bank_statement_line.xml
Comment (tien-ld) [inline account_reconcile_oca/views/account_bank_statement_line.xml]: shouldn't it be `active_id` instead of `id`

image

File: account_reconcile_oca/views/account_move_line.xml
Comment (tien-ld) [inline account_reconcile_oca/views/account_move_line.xml]: I think it should be `column_invisible`, in version 17 invisible only hides data, not the entire column

**Review by tien-ld — CHANGES_REQUESTED** [review]

**Comment by anhito** [comment]
please dispaly reconcile model button  rule by type "Button to generate counterpart entry" like version 16.
when first time reconcile page, nothing bank statement line selected

File: account_reconcile_oca/models/account_bank_statement_line.py
Comment (etobella) [reply account_reconcile_oca/models/account_bank_statement_line.py:673]: Why? This is a private function, no need of that...

File: account_reconcile_oca/views/account_bank_statement_line.xml
Comment (etobella) [reply account_reconcile_oca/views/account_bank_statement_line.xml]: Fixed

File: account_reconcile_oca/models/account_bank_statement_line.py
Comment (tien-ld) [reply account_reconcile_oca/models/account_bank_statement_line.py:673]: because in the function, it used `self.move_id.id` which will not work when `self` contains multiple records.

we need to use `ensure_one()` when the function involves operations with recordsets and it should only operate on a single record

So it might be:
```python
if not move_id:
    self.ensure_one()
```

**Comment by norlinhenrik** [comment]
@etobella What do you think about the suggestions from @tien-ld ?

**Comment by etobella** [comment]
I accepted two and I refused two with a comment. Or am I missing something?

**Comment by xtanuiha** [comment]
I have contributed translations for Simplified Chinese, could you please merge them, thank you!

[zh_CN.zip](https://github.com/OCA/account-reconcile/files/14916800/zh_CN.zip)

**Comment by pedrobaeza** [comment]
They should be added in Weblate once this is merged.

**Comment by norlinhenrik** [comment]
@xtanuiha You are welcome to review the PR and approve it if you think it is ready to merge.

File: account_reconcile_oca/static/src/js/reconcile/reconcile_controller.esm.js
Comment (xtanuiha) [inline account_reconcile_oca/static/src/js/reconcile/reconcile_controller.esm.js:1]: I noticed that the JavaScript files within Odoo 17’s first-party modules now all have a .js suffix and are not required to end with .esm.js. Would it be better to change the file suffixes to .js?

File: account_reconcile_oca/views/account_account_reconcile.xml
Comment (xtanuiha) [inline account_reconcile_oca/views/account_account_reconcile.xml:19]: The styles of these buttons are not very appealing; making these modifications would be an improvement.
```xml
<div class="d-flex gap-1">
```

File: account_reconcile_oca/static/src/js/reconcile/reconcile_controller.esm.js
Comment (etobella) [reply account_reconcile_oca/static/src/js/reconcile/reconcile_controller.esm.js:1]: This is how it is managed inside OCA

File: account_reconcile_oca/models/account_bank_statement_line.py
Comment (etobella) [reply account_reconcile_oca/models/account_bank_statement_line.py:673]: No need to ensure_one on private functions. That should be managed on public functions.

**Review by hbrunn — APPROVED** [review]

**Review by xtanuiha — APPROVED** [review]

**Pull request merged**
